script.sh

```diff
@@ -1,29 +1,105 @@
 #!/bin/bash
 
+# Fail fast on errors, unset variables, and failures in piped commands
+set -Eeuo pipefail
+
 cd "${GITHUB_WORKSPACE}/${INPUT_WORKING_DIRECTORY}" || exit
 
-TEMP_PATH="$(mktemp -d)"
-PATH="${TEMP_PATH}:$PATH"
+echo '::group::Preparing'
+  unameOS="$(uname -s)"
+  case "${unameOS}" in
+    Linux*)     os=linux;;
+    Darwin*)    os=darwin;;
+    CYGWIN*)    os=windows;;
+    MINGW*)     os=windows;;
+    MSYS_NT*)   os=windows;;
+    *)          echo "Unknown system: ${unameOS}" && exit 1
+  esac
+
+  unameArch="$(uname -m)"
+  case "${unameArch}" in
+    x86*)      arch=amd64;;
+    aarch64*)  arch=arm;;
+    *)         echo "Unsupported architecture: ${unameArch}" && exit 1
+  esac
+
+  TEMP_PATH="$(mktemp -d)"
+  echo "Detected ${os} running on ${arch}, will install tools in ${TEMP_PATH}"
+  REVIEWDOG_PATH="${TEMP_PATH}/reviewdog"
+  TFLINT_PATH="${TEMP_PATH}/tflint"
+
+  if [[ -z "${INPUT_TFLINT_VERSION}" ]] || [[ "${INPUT_TFLINT_VERSION}" == "latest" ]]; then
+    echo "Looking up the latest tflint version ..."
+    tflint_version=$(curl --silent --show-error --fail --location "https://api.github.com/repos/terraform-linters/tflint/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+  else
+    tflint_version=${INPUT_TFLINT_VERSION}
+  fi
+
+  if [[ -z "${TFLINT_PLUGIN_DIR:-}" ]]; then
+    export TFLINT_PLUGIN_DIR="${TFLINT_PATH}/.tflint.d/plugins"
+    mkdir -p "${TFLINT_PLUGIN_DIR}"
+    echo "TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR}" >> "${GITHUB_ENV}"
+  else
+    echo "Found pre-configured TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR}"
+  fi
+echo '::endgroup::'
+
+echo "::group::🐶 Installing reviewdog (${REVIEWDOG_VERSION}) ... https://github.com/reviewdog/reviewdog"
+  curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${REVIEWDOG_PATH}" "${REVIEWDOG_VERSION}" 2>&1
 
-echo '::group::🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog'
-curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${TEMP_PATH}" "${REVIEWDOG_VERSION}" 2>&1
+  echo "${REVIEWDOG_PATH}" >> "${GITHUB_PATH}"
```

Comment:
This action should not install (overwrite) global reviewdog.
Can you revert this?

Suppose many reviewdog actions install and overwrite reviewdog in global PATH, it could introduce needless complexity and surpriseto users.

Reply:
Done in https://github.com/reviewdog/action-tflint/pull/35/commits/3306f7cd1ad5e6b59d9554c03c498d73ccdbb835